feature/gribi/ate_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go

```diff
@@ -306,15 +307,20 @@ func testRecursiveIPv4Entry(t *testing.T, args *testArgs) {
 			t.Errorf("next-hop index is incorrect: got %v, want %v", got, want)
 		}
 		nh := gnmi.Get(t, args.dut, gnmi.OC().NetworkInstance(*deviations.DefaultNetworkInstance).Afts().NextHop(nhIndexInst).State())
-		if got, want := nh.GetIpAddress(), atePort2.IPv4; got != want {
-			t.Errorf("next-hop is incorrect: got %v, want %v", got, want)
+		// for devices that return  the nexthop with resolving it recursively. For a->b->c the device returns c
+		if got := nh.GetIpAddress(); got != atePort2.IPv4 {
```

Comment:
minor: how about `if got := nh.GetIpAddress(); got != atePort2.IPv4 && got != ateIndirectNH`?  (you can merge the comment together too.)